enclave/core/oelog.c

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/bits/safecrt.h>
+#include <openenclave/bits/safemath.h>
+#include <openenclave/bits/types.h>
+#include <openenclave/bits/log-enc.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/enclavelibc.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/sgxtypes.h>
+#include <openenclave/internal/utils.h>
+
+oe_result_t oe_log(uint8_t level, const char* module, const char *fmt, ...)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    oe_log_args_t* args = NULL;
+
+    if (!module || !fmt)
+        goto done;
+
+    if (!(args = oe_host_malloc(sizeof(oe_log_args_t))))
+    {
+        result = OE_OUT_OF_MEMORY;
+        goto done;
+    }
+
+    args->level = level;
+    if (oe_strncpy_s(
+            args->module,
+            sizeof(char) * OE_LOG_MODULE_LEN_MAX,
+            module, oe_strlen(module)) != OE_OK)
+        goto done;
```

Comment:
The preferred coding style is OE_CHECK or OE_RAISE. 
It ensures that the errors are propagated all the way up.